### PR TITLE
feat(node_framework): Support Proof Data Handler in the framework

### PR DESCRIPTION
**Pull request by AnastasiiaVashchuk**
## What ❔
- Adds `Proof data handler` implementation, i. e. `ProofDataHandlerLayer` and `ProofDataHandlerTask`, in the same way it was done for `StateKeeper ` [here](https://github.com/matter-labs/zksync-era/pull/1043/files#diff-8ff8babf7b83c79dbf96f4998cf71d888beaeb265c7ce33192b0bb0c808f662b).

Current "external" point of view for the framework(the only thing changed is that we add 
`.add_proof_data_handler_layer()?` after `.add_eth_watch_layer()?`):

```rust
fn main() -> anyhow::Result<()> {
    #[allow(deprecated)] // TODO (QIT-21): Use centralized configuration approach.
    let log_format = vlog::log_format_from_env();
    let _guard = vlog::ObservabilityBuilder::new()
        .with_log_format(log_format)
        .build();

    MainNodeBuilder::new()
        .add_pools_layer()?
        .add_fee_input_layer()?
        .add_object_store_layer()?
        .add_metadata_calculator_layer()?
        .add_state_keeper_layer()?
        .add_eth_watch_layer()?
        .add_proof_data_handler_layer()?
        .add_healthcheck_layer()?
        .build()
        .run()?;

    Ok(())
}
```

### CODE REVIEW

File: core/node/node_framework/src/implementations/layers/proof_data_handler.rs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+impl WiringLayer for ProofDataHandlerLayer {
+    fn layer_name(&self) -> &'static str {
+        "proof_data_handler_layer"
+    }
+
+    async fn wire(self: Box<Self>, context: ServiceContext<'_>) -> Result<(), WiringError> {
+        let pool_resource = context.get_resource::<MasterPoolResource>().await?;
+        let main_pool = pool_resource.get().await.unwrap();
+
+        let object_store = context.get_resource::<ObjectStoreResource>().await.unwrap();
```
Comment (popzxc) [inline core/node/node_framework/src/implementations/layers/proof_data_handler.rs:47]: I don't think we should unwrap here.

File: core/node/node_framework/src/implementations/layers/proof_data_handler.rs [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    fn name(&self) -> &'static str {
+        "proof_data_handler"
+    }
+
+    async fn run(self: Box<Self>, stop_receiver: StopReceiver) -> anyhow::Result<()> {
+        proof_data_handler::run_server(
+            self.proof_data_handler_config
+                .clone()
+                .context("proof_data_handler_config")?,
+            self.contracts_config.clone().context("contracts_config")?,
```
Comment (popzxc) [inline core/node/node_framework/src/implementations/layers/proof_data_handler.rs:79]: These lines don't seem right.

**Review by popzxc — COMMENTED** [review]
Looks like CI is failing due to some compile errors.

File: core/node/node_framework/src/implementations/layers/proof_data_handler.rs (hunk 1 above)
Comment (AnastasiiaVashchuk) [reply core/node/node_framework/src/implementations/layers/proof_data_handler.rs:47]: `await?` then?

File: core/node/node_framework/src/implementations/layers/proof_data_handler.rs (hunk 2 above)
Comment (AnastasiiaVashchuk) [reply core/node/node_framework/src/implementations/layers/proof_data_handler.rs]: indeed.

**Pull request merged**